state/src/applicator/task_queue.rs

```diff
@@ -162,7 +160,7 @@ impl StateApplicator {
         // was paused
         tx.resume_task_queue(&key)?;
         let task = tx.pop_task(&key)?.expect("expected preemptive task");
-        if let Some(mut t) = HistoricalTask::from_queued_task(task, key) {
+        if let Some(mut t) = HistoricalTask::from_queued_task(task.clone(), key) {
```

Comment:
similar to `pop_task`, should we mutate the task state outside of this conditional so that the updated state is published to the system bus correctly?

Reply:
yes, good catch, done